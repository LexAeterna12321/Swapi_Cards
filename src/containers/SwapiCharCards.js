import React, { Component } from "react";
import axios from "axios";
import SwapiCharCard from "../components/SwapiCharCard";
import Avatar from "../components/Avatar";
import Description from "../components/Description";
import { Container, Cards } from "../components/styledComponents";
class SwapiCharCards extends Component {
  state = { chars: [] };
  apiURL = `https://swapi.co/api`;

  componentDidMount() {
    this.getCharsPersonalData().then(chars =>
      this.setState({ chars: [...chars] })
    );
  }

  getCharsPersonalData = async () => {
    const response = await axios.get(`${this.apiURL}/people`);
    const {
      data: { results }
    } = response;
    const chars = results.slice(0, 2);
    return chars;
  };

  render() {
    const { chars } = this.state;
    return (
      <Container>
        <Cards>
          {chars.map(char => {
            const { name, birth_year, gender, url } = char;

            return (
              <SwapiCharCard key={url}>
                <Avatar name={name} />
                <Description
                  name={name}
                  birth_year={birth_year}
                  gender={gender}
                />
              </SwapiCharCard>
            );
          })}
        </Cards>
      </Container>
    );
  }
}

export default SwapiCharCards;
