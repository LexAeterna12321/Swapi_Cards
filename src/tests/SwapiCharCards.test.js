import React from "react";
import SwapiCharCards from "../containers/SwapiCharCards";
import { shallow } from "enzyme";

let wrapper;
describe("SwapiCharCard component render / Swapi API work check", () => {
  wrapper = shallow(<SwapiCharCards />);
  it("expects to render SwapiCharCard component", () => {
    expect(shallow(<SwapiCharCards />)).toMatchSnapshot();
  });

  it("gets swapi characters data", done => {
    expect.assertions(2);
    wrapper
      .instance()
      .getCharsPersonalData()
      .then(res => {
        expect(res).toEqual([...res]);
        expect(res).toHaveLength(2);
        done();
      });
  });
});
