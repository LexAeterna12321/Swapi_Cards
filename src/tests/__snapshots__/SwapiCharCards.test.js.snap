// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapiCharCard component render / Swapi API work check expects to render SwapiCharCard component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SwapiCharCards />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 5%;
  align-items: center;
  justify-items: center;
  border: 1px solid gold;
  box-shadow: inset 0 0 5px 3px gold;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  height: 100vh;
  background: #c2c2c2;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 5%;
  align-items: center;
  justify-items: center;
  border: 1px solid gold;
  box-shadow: inset 0 0 5px 3px gold;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  height: 100vh;
  background: #c2c2c2;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
